src/transform/transform.ts

```diff
@@ -481,3 +495,66 @@ function generateResponseTypeName(schema: Schema): Language {
     responseType: true,
   }
 }
+
+function annotateDiscriminatedTypes(obj: ObjectSchema): ObjectSchema | undefined {
+  if (obj.language.go!.polymorphicInterfaces !== undefined) {
+    // this hierarchy of discriminated types has already been processed
+    return;
+  }
+  // we have a type in the hierarchy of polymorphic types, it can be one of three things
+  // 1. root - no parent types, only child types
+  // 2. intermediate root - has a parent and also has children (salmon in the test server)
+  // 3. child - has parent and no children
+  //
+  // for cases #1 and #2 we need to generate an interface type, and for
+  // case #2 the generated interface must also contain the parent interface
+  // for case #3 all that's required is to generate the marker method on
+  // the child type(s) for the applicable interface.
+
+  // walk to the root
+  let root = obj;
+  while (true) {
+    if (!root.parents) {
+      break;
+    }
+    for (const parent of values(root.parents?.immediate)) {
+      // there can be parents that aren't part of the hierarchy.
+      // e.g. if type Foo is in a DictionaryOfFoo, then one of
+      // Foo's parents will be DictionaryOfFoo which we ignore.
+      if (isObjectSchema(parent)) {
+        root = parent;
+      }
+    }
+    // fail-safe to prevent infinite loop due to a bug somewhere...
+    if (root === obj) {
```

Comment:
Is this bug pointing to a swagger that expresses the parent and the child as the same? OR what happened here? Just to know.

Reply:
Nothing actually happened here, it's a preventative measure.  E.g. perhaps my loop isn't correct in some obscure case, or something changes in M4 etc, it will prevent the generator from just hanging.

Reply:
Ah I see.